Repository/RepoTreeView.cpp

```diff
@@ -143,6 +146,64 @@ void RepoTreeView::onRepoDeactivated(RM::Repo* repo)
     }
 }
 
+/**
+ * @brief This slot is called, when a single fetch operation finishes.
+ */
+void RepoTreeView::fetchOperationFinished()
+{
+    Git::FetchOperation* op = qobject_cast<Git::FetchOperation*>( sender() );
+    Q_ASSERT( op );
+
+    Git::Result r( op->result() );
+    if ( !r ) {
+        QMessageBox::warning( this, tr("Operation failed."),
+                              tr("Failed to fetch repository '%1'.\nMessage: %2")
+                              .arg(op->repository().name()).arg(r.errorText())
+                              );
+    }
+
+    // delete the operation
+    op->deleteLater();
+}
+
+void RepoTreeView::onCtxFetchAll()
+{
+    Heaven::Action* action = qobject_cast< Heaven::Action* >( sender() );
```

Comment:
Actually this code does what it should do. Why should we remove that? :smile:

In this special case, the action is actually working and we add value by adding the unfinished fetch operation. The only thing is: A "fetch" action currently works just with unencrypted protocols and it doesn't show progress yet.

:zap: Returning a function without doing anything is the worst thing a programmer could code in released code. How would you react to buttons without a function? For an example, see the "Remotes" menu :smile:.

Btw.: How about providing a `GW_TODO` macro in libGitWrap, that produces an error `Git::Result` with error code GIT_EUSER and a message like this:
```text
The method 'Git::BestOperationEver::run()' in file:line is not implemented yet. Want to help out? Please contact us: <-maintainer->@macgitver.org.
```

Inspirational credits go to Blender :smile:.